### ISSUE TITLE
nvidia failed to compile in f22 beta

### ISSUE THREAD

**Issue by rudrab** [Bumblebee-Project/Bumblebee#657]
Hi,
I have reported a similar issue with f22-alpha, and was solved.
In my new f22-beta installation, the problem is back.

```
# tail -30 /var/log/nvidia-installer.log 
PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin

nvidia-installer command line:
    ./nvidia-installer
    --accept-license
    --silent
    --no-x-check
    --no-nouveau-check
    --no-recursion
    --opengl-libdir=lib64/nvidia-bumblebee
    --opengl-prefix=/usr
    --x-library-path=lib64/nvidia-bumblebee
    --x-prefix=/usr
    --x-module-path=/usr/lib64/nvidia-bumblebee/xorg/modules
    --compat32-libdir=lib/nvidia-bumblebee
    --compat32-prefix=/usr

Using built-in stream user interface
-> Detected 8 CPUs online; setting concurrency level to 8.
-> Tagging shared libraries with chcon -t textrel_shlib_t.
-> License accepted by command line option.
-> Installing NVIDIA driver version 346.47.
-> Would you like to register the kernel module sources with DKMS? This will allow DKMS to automatically build a new module, if you install a different kernel later. (Answer: No)
-> Performing CC sanity check with CC="/usr/bin/cc".
-> The CC version check failed:

The compiler used to compile the kernel (gcc 5.0) does not exactly match the current compiler (gcc 5.1).  The Linux 2.6 kernel module loader rejects kernel modules built with a version of gcc that does not exactly match that of the compiler used to build the running kernel.

If you know what you are doing you can either ignore the CC version check and continue installation, or abort installation, set the CC environment variable to the name of the compiler used to compile your kernel, and restart installation. (Answer: Abort installation)
ERROR: Installation has failed.  Please see the file '/var/log/nvidia-installer.log' for details.  You may find suggestions on fixing installation problems in the README available on the Linux driver download page at www.nvidia.com.

systemctl status -l bumblebee-nvidia.service 
● bumblebee-nvidia.service - Compiling NVidia Driver
   Loaded: loaded (/etc/systemd/system/bumblebee-nvidia.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Wed 2015-04-29 13:22:05 BST; 9min ago
  Process: 664 ExecStart=/usr/sbin/bumblebee-nvidia (code=exited, status=1/FAILURE)
 Main PID: 664 (code=exited, status=1/FAILURE)

Apr 29 13:21:49 roddur bumblebee-nvidia[664]: --force compile selected via /etc/sysconfig/nvidia/compile-nvidia-driver
Apr 29 13:21:53 roddur bumblebee-nvidia[664]: Building NVIDIA video drivers: Creating directory NVIDIA-Linux-x86_64-346.47
Apr 29 13:21:54 roddur bumblebee-nvidia[664]: Verifying archive integrity... OK
Apr 29 13:22:02 roddur bumblebee-nvidia[664]: Uncompressing NVIDIA Accelerated Graphics Driver for Linux-x86_64 346.47...........................................................................................................................................................................................................................................................
Apr 29 13:22:02 roddur bumblebee-nvidia[664]: patching file kernel/nv-pat.c
Apr 29 13:22:05 roddur bumblebee-nvidia[664]: [FAILED]
Apr 29 13:22:05 roddur systemd[1]: bumblebee-nvidia.service: main process exited, code=exited, status=1/FAILURE
Apr 29 13:22:05 roddur systemd[1]: Failed to start Compiling NVidia Driver.
Apr 29 13:22:05 roddur systemd[1]: Unit bumblebee-nvidia.service entered failed state.
Apr 29 13:22:05 roddur systemd[1]: bumblebee-nvidia.service failed.

# systemctl status -l bumblebeed.service 
● bumblebeed.service - Bumblebee C Daemon
   Loaded: loaded (/usr/lib/systemd/system/bumblebeed.service; enabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: exit-code) since Wed 2015-04-29 13:30:56 BST; 48s ago
  Process: 4418 ExecStart=/usr/sbin/bumblebeed (code=exited, status=1/FAILURE)
 Main PID: 4418 (code=exited, status=1/FAILURE)
   CGroup: /system.slice/bumblebeed.service

Apr 29 13:30:56 roddur systemd[1]: Unit bumblebeed.service entered failed state.
Apr 29 13:30:56 roddur systemd[1]: bumblebeed.service failed.
```

This is the status in my system.
Kindly help

**Comment by ArchangeGabriel** [comment]
So, first, I’ve changed your title, because this is not bumbebee that’s failing to compile, but nvidia.

Second, the issue is clearly stated in the log: the kernel was compiled with gcc5.0, but you have gcc5.1. That’s a minor but annoying issue, as a matter of fact gcc5.0 doesn’t exist because it was released as gcc5.1. So, as the log suggests, you should tell him to ignore the mismatch.

**Comment by rudrab** [comment]
Thanks for your reply. I am a enduser and installed bumblebee(-nvidia) from the rpm file as instructed in https://fedoraproject.org/wiki/Bumblebee.
I cannot recompile from source. 
May a rpm possible?

**Comment by ArchangeGabriel** [comment]
If you have a nvidia-installer.log, this means the compilation (the nvidia one, not talking about bumblebee or the kernel) happens on your computer. From what I understand, the RPM is responsible of the compilation, but I’m not an expert in RPM at all.

I’m calling @gsgatlin to help you on this.

**Comment by gsgatlin** [comment]
I think this will sort itself out once a new kernel update comes out for f22. But in the meantime I could not get it to work.

"dnf downgrade gcc" does not do anything in the f22 beta iso. You could go try to find the proper gcc  5.0 rpms by hand but its probably better to wait a few days. I will keep an eye on this in my test laptop.

**Comment by gsgatlin** [comment]
Hello.

I believe this has been fixed with kernel 4.0.1-300.fc22. Try updating and see if its working for you now.

Cheers.
